src/AzureClient/Extensions.cs

```diff
@@ -22,5 +30,96 @@ public static void AddAzureClient(this IServiceCollection services)
         {
             services.AddSingleton<IAzureClient, AzureClient>();
         }
+
+        /// <summary>
+        ///      Encapsulates a given <c>AzureClientError</c> as the result of an execution.
+        /// </summary>
+        /// <param name="azureClientError">
+        ///      The result of an IAzureClient API call.
+        /// </param>
+        public static ExecutionResult ToExecutionResult(this AzureClientError azureClientError) =>
+            new ExecutionResult
+            {
+                Status = ExecuteStatus.Error,
+                Output = azureClientError.ToDescription()
+            };
+
+        /// <summary>
+        ///     Returns the string value of the <c>DescriptionAttribute</c> for the given
+        ///     <c>AzureClientError</c> enumeration value.
+        /// </summary>
+        /// <param name="azureClientError"></param>
+        /// <returns></returns>
+        public static string ToDescription(this AzureClientError azureClientError)
+        {
+            var attributes = azureClientError
+                .GetType()
+                .GetField(azureClientError.ToString())
+                .GetCustomAttributes(typeof(DescriptionAttribute), false) as DescriptionAttribute[];
+            return attributes?.Length > 0 ? attributes[0].Description : string.Empty;
+        }
+
+        /// <summary>
+        ///      Encapsulates a given <c>AzureClientError</c> as the result of an execution.
+        /// </summary>
+        /// <param name="task">
+        ///      A task which will return the result of an IAzureClient API call.
+        /// </param>
+        public static async Task<ExecutionResult> ToExecutionResult(this Task<AzureClientError> task) =>
```

Comment:
It may be good to add the optional `ExecuteStatus` parameter here as well.

Reply:
This ends up calling the `AzureClientError.ToExecutionResult` extension, which always sets `ExecuteStatus.Error`. I'm not sure it makes sense to allow a different status for an `AzureClientError`?

---
In reply to: [427471505](https://github.com/microsoft/iqsharp/pull/144#discussion_r427471505) [](ancestors = 427471505)